sw/device/tests/otbn_randomness_test.c

```diff
@@ -38,16 +44,52 @@ static void print_uint256(otbn_t *ctx, const otbn_addr_t var,
            data[5], data[4], data[3], data[2], data[1], data[0]);
 }
 
+void initialize_clkmgr(void) {
+  mmio_region_t addr = mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR);
+  CHECK_DIF_OK(dif_clkmgr_init(addr, &clkmgr));
+
+  // Get initial hint and enable for OTBN clock and check both are enabled.
+  dif_toggle_t clock_hint_state;
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_get_hint(&clkmgr, otbn_clock,
+                                                  &clock_hint_state));
+  CHECK(clock_hint_state == kDifToggleEnabled);
+  CLKMGR_TESTUTILS_CHECK_CLOCK_HINT(&clkmgr, otbn_clock, kDifToggleEnabled);
+
+  // While OTBN is not in use, set disabled hint for OTBN clock
+  // then check clock state, set back to enabled and check again.
+  // Note: disabled means the clock can be disabled.
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, otbn_clock,
+                                                  kDifToggleDisabled));
+  CLKMGR_TESTUTILS_CHECK_CLOCK_HINT(&clkmgr, otbn_clock, kDifToggleDisabled);
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, otbn_clock,
+                                                  kDifToggleEnabled));
+  CLKMGR_TESTUTILS_CHECK_CLOCK_HINT(&clkmgr, otbn_clock, kDifToggleEnabled);
+}
+
 bool test_main() {
   entropy_testutils_boot_mode_init();
 
+  initialize_clkmgr();
+
   // Initialize
   otbn_t otbn_ctx;
   CHECK(otbn_init(&otbn_ctx, mmio_region_from_addr(
                                  TOP_EARLGREY_OTBN_BASE_ADDR)) == kOtbnOk);
   CHECK(otbn_load_app(&otbn_ctx, kOtbnAppCfiTest) == kOtbnOk);
 
+  /* Start an OTBN operation.
+   * Write the OTBN clk hint to 0 within clkmgr and verify that the OTBN clk
+   * hint status within clkmgr reads 0 (OTBN is not idle), before the operation
+   * is complete.
+   */
   CHECK(otbn_execute(&otbn_ctx) == kOtbnOk);
+
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, otbn_clock,
+                                                  kDifToggleDisabled));
+  CLKMGR_TESTUTILS_CHECK_CLOCK_HINT(&clkmgr, otbn_clock, kDifToggleEnabled);
+  CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(&clkmgr, otbn_clock,
```

Comment:
If I don't enable the clock hint again the test blocks busy waint otbn reports idle. 
It seems that the otbn enters in idle and the clkmgr disables its clock before a signal propagates.  
![image](https://user-images.githubusercontent.com/92785679/155187330-9290ff93-ed04-4eb5-8761-d595c279e50f.png)
![image](https://user-images.githubusercontent.com/92785679/155187425-546123e9-7570-47d9-8fd9-46f8d63d8da9.png)

Reply:
Ah! Yeah, that would happen. I think you should probably see an interrupt, though?

Reply:
I haven't tried with SW yet, but looking at the traces I think that the interrupt won't work, right?
![image](https://user-images.githubusercontent.com/92785679/155211827-b4a0a6e3-d601-4caf-87e5-d6f91511a59f.png)

Reply:
Nop, the `intr_state` goes to `1`, but `intr_done_o` doesn't change and the core never leaves the `wfi` instruction. 

![image](https://user-images.githubusercontent.com/92785679/155342847-51acc84c-0876-425f-b965-e7abd66b538c.png)

Reply:
Oh dear! That's arguably a proper problem with the system! @tjaychen/@msfschaffner: Is that what you'd expect? Should we be tweaking OTBN so that it sends out its interrupt before telling clkmgr that it's idle? (@GregAC: thoughts?)

Reply:
ahhh... i think technically we do need to ensure the interrupt can be generated before asking the clocks to turn off.
I guess this is happening because the interrupt output is flopped, and that one extra cycle we just happened to not have. 

I think there are a few solutions in my mind.

1. update otbn to keep idle asserted a bit longer
2. update clkmgr to not immediately shut off, but allow some breathing room

On the other hand, due to security reasons, we may need to create a full handshaking on the idle / clock gate handling, so this might just get naturally fixed.